### PR TITLE
[cuda] correct env variables for newer cuda versions

### LINKED ISSUE #11976: [cuda] build failure
**Host Environment**

Windows 10 
Cuda 11

**To Reproduce**
 .\vcpkg install cuda

**Failure logs** 
Error: Building package cuda:x86-windows failed with: BUILD_FAILED
  Package: cuda:x86-windows
  Vcpkg version: 2020.02.04-nohash

**Additional context**
Add any other context about the problem here, such as what you have already tried to resolve the issue.

### PR DESCRIPTION
**Pull request by Neumann-A**
A bit of version updating for newer cuda versions

### CODE REVIEW

**Comment by JackBoosY** [comment]
Does this PR fixes #11976?

**Review by JackBoosY — APPROVED** [review]

**Comment by Neumann-A** [comment]
@JackBoosY: At least for me:
```
Starting package 1/19: cuda:x86-windows
Building package cuda[core]:x86-windows...
-- Found CUDA V11.0.167
-- Performing post-build validation
-- Performing post-build validation done
Building package cuda[core]:x86-windows... done
Installing package cuda[core]:x86-windows...
Installing package cuda[core]:x86-windows... done
```

**Comment by JackBoosY** [comment]
CUDA worked very well in the past, so I think this is caused by CUDA or findCUDA.cmake update. Do we need to judge based on the version of CUDA?

**Comment by Neumann-A** [comment]
I think it is related to updating visual studio which somehow wipes `CUDA_PATH`
After upgrading CUDA to 11 I got the following env vars now:
```
CUDA_PATH=C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v11.0
CUDA_PATH_V10_2=C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v10.2
CUDA_PATH_V11_0=C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v11.0
```
CUDA also worked for me two months or so ago and suddenly broke. So it is highly likely that a VS upgrade removed `CUDA_PATH` and only left the versioned `CUDA_PATH_V_XX_X`.

**Comment by JackBoosY** [comment]
Anyway, this changes LGTM.

**Comment by Voskrese** [comment]
new rules are now in effect))

 ~~Version: 10.1-2~~

```
Version: 10.1
 Port-Version: 2
```

File: ports/cuda/CONTROL [hunk 1]
```diff
@@ -1,4 +1,4 @@
 Source: cuda
-Version: 10.1-1
+Version: 10.1-2
```
Comment (JackBoosY) [inline ports/cuda/CONTROL:2]:
```suggestion
Version: 10.1
Port-Version: 2
```

**Review by JackBoosY — CHANGES_REQUESTED** [review]

**Review by JackBoosY — APPROVED** [review]

**Comment by Neumann-A** [comment]
@JackBoosY: We might need to define CUDA_PATHS  on a more global level so that dependent ports can actually leverage the additional search paths but that is something for a different PR

**Comment by JackBoosY** [comment]
@Neumann-A  think we should add it to `vcpkg_common_definitions` or add a vcpkg function to provide it.

**Comment by JackBoosY** [comment]
Let my colleague check this PR.

**Comment by cenit** [comment]
CUDA_PATH should suffice tbh and it is working for cuda 11 for me without this PR :/ (and I also have vs fully up to date...)

**Comment by Neumann-A** [comment]
Another possibility is that a Windows Update killed the `CUDA_PATH` environment variable. Something remove it. nvcc was still on path for me so the `PATH` variable has not been touched

**Comment by BillyONeal** [comment]
The version check here looks bogus and we are still looking for a better way to publish the detected CUDA information for other ports, so I think we should merge this as is but we don't want to close the door on fixing that the right way.

Also this version check looks wrong https://github.com/microsoft/vcpkg/pull/12244/files#diff-efb436f7552399e0b189b636d44230f0R81

**Comment by cenit** [comment]
I remain on my idea that CUDA_PATH has to be defined in an environment for CUDA to be *really* working. It does not matter that this PR looks like fixing something, it’s just hiding problems which will reappear downwards. So, I would have left only CUDA_PATH for finding cuda. But again, it’s already merged, so be it.
The mood might be because today’s round of updates broke many ports on me, maybe...

**Comment by StarGate-One** [comment]
@Neumann-A I lost CUDA 11 (along with some other VS extension) in both my VS2017 and VS2019 with the latest upgrades of VS. I had to reinstall the CUDA package to get it back in both versions. 
I do not know why some VS upgrades (especially like a 16.6.3 to 16.6.4) wipe out some extensions but keep others????

**Pull request merged**